### PR TITLE
Add unit and functional tests to mmsign and reposign commands

### LINKED ISSUE #6684: Add tests for nuget mssign
There are no end to end scenarios covered for the `sign` and `reposign` command in the mssign exe.

### PR DESCRIPTION
**Pull request by PatoBeltran**
Fixes https://github.com/NuGet/Home/issues/6684

This PR adds unit and functional tests for NuGet `mssign` and `reposing` commands. Also includes some fixes for a better error experience and better testability in the `reposing` and `mssign`, as well as a typo in the `mssign` usage description.

### CODE REVIEW

File: src/NuGet.Clients/NuGet.MSSigning.Extensions/MSSignAbstract.cs [hunk 1]
```diff
@@ -59,20 +59,27 @@ protected CngKey GetPrivateKey(X509Certificate2 cert)
...
-            var provider = new CngProvider(CSPName);
-            var cngkey = CngKey.Open(KeyContainer, provider, CngKeyOpenOptions.MachineKey);
-
-            if (cngkey == null)
+            try
             {
-                throw new InvalidOperationException(NuGetMSSignCommand.MSSignCommandNoCngKeyException);
-            }
+                var provider = new CngProvider(CSPName);
+                var cngkey = CngKey.Open(KeyContainer, provider, CngKeyOpenOptions.MachineKey);
```
Comment (dtivel) [inline src/NuGet.Clients/NuGet.MSSigning.Extensions/MSSignAbstract.cs]: `CngKey` implements `IDisposable`, so you should call `Dispose()` in exceptional paths where `cngKey` may not be `null`:

```
CngKey cngKey = null;

try
{
    // Do stuff

    if (cngkey.AlgorithmGroup != CngAlgorithmGroup.Rsa)
    {
        cngKey.Dispose();

        throw new InvalidOperationException(NuGetMSSignCommand.MSSignCommandInvalidCngKeyException);
    }
}
catch (CryptographicException)
{
    cngKey?.Dispose();
}
```

File: src/NuGet.Clients/NuGet.MSSigning.Extensions/NuGetMSSignCommand.resx [hunk 2]
```diff
@@ -194,7 +194,7 @@ nuget mssign MyPackage.nupkg -Timestamper https://foo.bar -CertificateFile foo.p
   <data name="RepoSignCommandUsageExamples" xml:space="preserve">
     <value>nuget reposign MyPackage.nupkg -Timestamper http://timestamper.test -CertificateFile certificates.p7b -CSPName "Cryptographic Service Provider" -KeyContainer 4003d786-cc37-4004-bfdf-c4f3e8ef9b3a -CertificateFingerprint 8599ADD1C62EE42E315EA887FF60908B7A7C6E9B -V3ServiceIndexUrl https://v3.index.test
 
-nuget reposign MyPackage.nupkg -Timestamper http://timestamper.test -CertificateFile certificates.p7b -CSPName "Cryptographic Service Provider" -KeyContainer 4003d786-cc37-4004-bfdf-c4f3e8ef9b3a -CertificateFingerprint 8599ADD1C62EE42E315EA887FF60908B7A7C6E9B -OutputDirectory .\..\Signed -PackageOwners owner1;owner2 -V3ServiceIndexUrl https://v3.index.test</value>
+nuget reposign MyPackage.nupkg -Timestamper http://timestamper.test -CertificateFile certificates.p7b -CSPName "Cryptographic Service Provider" -KeyContainer 4003d786-cc37-4004-bfdf-c4f3e8ef9b3a -CertificateFingerprint 8599ADD1C62EE42E315EA887FF60908B7A7C6E9B -OutputDirectory .\..\Signed -PackageOwners "owner1;owner2" -V3ServiceIndexUrl https://v3.index.test</value>
```
Comment (dtivel) [inline src/NuGet.Clients/NuGet.MSSigning.Extensions/NuGetMSSignCommand.resx:197]: This is the typo?  When are quotation marks required?

File: test/TestUtilities/Test.Utility/Signing/SigningTestUtility.cs [hunk 3]
```diff
@@ -492,7 +492,8 @@ public static SignedCms GenerateRepositoryCountersignedSignedCms(X509Certificate
         /// </summary>
         public static X509Certificate2 GetPublicCert(X509Certificate2 cert)
         {
-            return new X509Certificate2(cert.Export(X509ContentType.Cert));
+            var pass = new Guid().ToString();
```
Comment (dtivel) [inline test/TestUtilities/Test.Utility/Signing/SigningTestUtility.cs]: password

File: test/TestUtilities/Test.Utility/Signing/SigningTestUtility.cs [hunk 4]
```diff
@@ -501,7 +502,7 @@ public static X509Certificate2 GetPublicCert(X509Certificate2 cert)
         public static X509Certificate2 GetPublicCertWithPrivateKey(X509Certificate2 cert)
         {
             var pass = new Guid().ToString();
```
Comment (dtivel) [inline test/TestUtilities/Test.Utility/Signing/SigningTestUtility.cs]: password

File: test/TestUtilities/Test.Utility/Signing/SigningTestUtility.cs [hunk 5]
```diff
@@ -492,7 +492,8 @@ public static SignedCms GenerateRepositoryCountersignedSignedCms(X509Certificate
         /// </summary>
         public static X509Certificate2 GetPublicCert(X509Certificate2 cert)
         {
-            return new X509Certificate2(cert.Export(X509ContentType.Cert));
+            var pass = new Guid().ToString();
+            return new X509Certificate2(cert.Export(X509ContentType.Cert), pass, X509KeyStorageFlags.Exportable);
```
Comment (dtivel) [inline test/TestUtilities/Test.Utility/Signing/SigningTestUtility.cs]: I think you can revert this change.

File: test/NuGet.Clients.FuncTests/NuGet.MSSigning.Extensions.FuncTest/Commands/MSSignCommandTests.cs [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+                var signCommand = new MSSignCommand
+                {
+                    Console = mockConsole.Object,
+                    Timestamper = timestampUri,
+                    CertificateFile = @"\\wrong\cert\path.pfx",
+                    CSPName = test.CertificateCSPName,
+                    KeyContainer = test.CertificateKeyContainer,
+                    CertificateFingerprint = test.Cert.Thumbprint,
+                };
+                signCommand.Arguments.Add(@"\\path\to\package.nupkg");
```
Comment (dtivel) [inline test/NuGet.Clients.FuncTests/NuGet.MSSigning.Extensions.FuncTest/Commands/MSSignCommandTests.cs]: I recommend against using UNC paths in any of these tests.

First, the test only succeeds on the assumption that the paths do not exist.  It's probably a reasonable assumption here, but you can't guarantee over time that these paths will never exist.

Second, NuGet will try to access these paths and fail after basically timing out.  That'll be 5-10 seconds.  That'll make the test 5-10 seconds longer than it needs to be to get the job done.  Try it out:

```PowerShell
PS C:\> $stopwatch = [System.Diagnostics.Stopwatch]::StartNew()
>> dir \\wrong\cert\path.pfx
>> Write-Host $stopwatch.Elapsed
dir : Cannot find path '\\wrong\cert\path.pfx' because it does not exist.
At line:2 char:1
+ dir \\wrong\cert\path.pfx
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (\\wrong\cert\path.pfx:String) [Get-ChildItem], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetChildItemCommand

00:00:07.8422013
```

Instead, create a local path you know does not exist.  You can use `TestDirectory` to do this.

**Review by dtivel — APPROVED** [review]

File: src/NuGet.Clients/NuGet.MSSigning.Extensions/NuGetMSSignCommand.resx (hunk 2 above)
Comment (PatoBeltran) [reply src/NuGet.Clients/NuGet.MSSigning.Extensions/NuGetMSSignCommand.resx:197]: I noticed that without quotation marks on the list, the command didn't parse al the owners correctly, that's why I added this
